test/BufferedIterator-test.js

```diff
@@ -30,7 +30,7 @@ describe('BufferedIterator', () => {
   describe('A BufferedIterator without arguments', () => {
     let iterator;
     before(() => {
-      iterator = new BufferedIterator();
+      iterator = new BufferedIterator({ autoStart: true });
```

Comment:
This is where this change will get a bit annoying, but I think it's important that the test structure remains the same. So first "without arguments" (so don't pass anything) and then the other cases.

(Note that the description and code are out of sync too.)